packages/contracts-bedrock/src/cannon/PreimageOracle.sol

```diff
@@ -3,18 +3,77 @@ pragma solidity 0.8.15;
 
 import { IPreimageOracle } from "./interfaces/IPreimageOracle.sol";
 import { PreimageKeyLib } from "./PreimageKeyLib.sol";
+import { LibKeccak } from "@lib-keccak/LibKeccak.sol";
 import "./libraries/CannonErrors.sol";
+import "./libraries/CannonTypes.sol";
 
 /// @title PreimageOracle
 /// @notice A contract for storing permissioned pre-images.
+/// @custom:attribution Solady <https://github.com/Vectorized/solady/blob/main/src/utils/MerkleProofLib.sol#L13-L43>
+/// @custom:attribution Beacon Deposit Contract / <0x00000000219ab540356cbb839cbe05303d7705fa>
 contract PreimageOracle is IPreimageOracle {
+    ////////////////////////////////////////////////////////////////
+    //                         Constants                          //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice The depth of the keccak256 merkle tree. Supports up to 65,536 keccak blocks, or ~8.91MB preimages.
+    uint256 public constant KECCAK_TREE_DEPTH = 16;
```

Comment:
Nice bump to depth `16`, I think this should provide sufficient confidence given our rough size buffer estimate.